torch/csrc/distributed/rpc/tensorpipe_agent.h

```diff
@@ -18,6 +18,20 @@ struct TensorPipeRpcBackendOptions : public RpcBackendOptions {
   std::map<std::string, worker_id_t> workerNameToId;
 };
 
+// Struct to track the network source metrics
+struct NetworkSourceInfo {
+  worker_id_t srcRank;
+  std::vector<uint8_t> srcMachineAddr;
+};
+
+// Struct to track aggregated network metrics
+struct AggregatedNetworkData {
+  uint64_t numCalls{0};
+  uint64_t totalSentBytes{0};
+  uint64_t totalRecvBytes{0};
```

Comment:
Maybe it makes sense to split these between payloads and tensors, since they are handled quite differently? If this will be used to drive decisions on how to optimize TensorPipe internal it would be useful to know...